src/libraries/System.Private.CoreLib/src/System/IO/Strategies/AsyncWindowsFileStreamStrategy.ValueTaskSource.cs

```diff
@@ -32,7 +32,7 @@ internal ValueTaskSource(AsyncWindowsFileStreamStrategy strategy)
             {
                 _strategy = strategy;
                 _source.RunContinuationsAsynchronously = true;
-                _preallocatedOverlapped = new PreAllocatedOverlapped(s_ioCallback, this, null);
+                _preallocatedOverlapped = PreAllocatedOverlapped.UnsafeCreate(s_ioCallback, this, null);
```

Comment:
Just to confirm - this change in behavior is intentional, correct? It looks like we're expecting that the continuations attached to the VTS have already captured their ECs correctly, so it's redundant (and inefficient) for us to capture the EC here. But wanted to make sure my understanding was correct.

(Same comment with similar lines in other partial files here.)

Reply:
> But wanted to make sure my understanding was correct.

It is.  The only way someone would observe a change in behavior here is if they manually used UnsafeAwaitOnCompleted to hook up their own continuation manually, in which case they're asking not to have ExecutionContext flowed.